From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Mon, 14 Oct 2024 07:34:38 +0000
Subject: [PATCH] revert-upstream: Revert clearing javascript JIT site settings

The bugfix upstream made is intended to clear the javascript JIT site
settings which was wired to a different site settings UI intended to
disable javascript optimizers instead of javascript JIT. Since the
said setting is exposed in Vanadium, do not clear javascript JIT site
settings.
---
 .../core/browser/content_settings_default_provider.cc       | 6 ++++++
 .../core/browser/content_settings_pref_provider.cc          | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/components/content_settings/core/browser/content_settings_default_provider.cc b/components/content_settings/core/browser/content_settings_default_provider.cc
index 0dd1396ad557e..16a3f55c59871 100644
--- a/components/content_settings/core/browser/content_settings_default_provider.cc
+++ b/components/content_settings/core/browser/content_settings_default_provider.cc
@@ -87,8 +87,10 @@ constexpr char kObsoleteTopLevelTpcdOriginTrialDefaultPref[] =
 // except via enterprise policy, so it is temporarily cleaned up here to revert
 // it to its default value.
 // TODO(https://crbug.com/367181093): clean this up.
+#if !BUILDFLAG(IS_ANDROID)
 constexpr char kBug364820109AlreadyWorkedAroundPref[] =
     "profile.did_work_around_bug_364820109_default";
+#endif  // !BUILDFLAG(IS_ANDROID)
 constexpr char kLocalNetworkAccessMigrateDefaultValuePref[] =
     "profile.default_content_setting_values.has_migrated_local_network_access";
 constexpr char kObsoleteTrackingProtectionDefaultPref[] =
@@ -165,8 +167,10 @@ void DefaultProvider::RegisterProfilePrefs(
   registry->RegisterIntegerPref(kObsoletePrivateNetworkGuardDefaultPref, 0);
 
 #if !BUILDFLAG(IS_IOS)
+#if !BUILDFLAG(IS_ANDROID)
   // TODO(https://crbug.com/367181093): clean this up.
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
+#endif  // !BUILDFLAG(IS_ANDROID)
 #endif  // !BUILDFLAG(IS_IOS)
 }
 
@@ -424,8 +428,10 @@ void DefaultProvider::DiscardOrMigrateObsoletePreferences() {
   prefs_->ClearPref(kObsoletePrivateNetworkGuardDefaultPref);
 
 #if !BUILDFLAG(IS_IOS)
+#if !BUILDFLAG(IS_ANDROID)
   // TODO(https://crbug.com/367181093): clean this up.
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
+#endif  // !BUILDFLAG(IS_ANDROID)
 #endif  // !BUILDFLAG(IS_IOS)
 }
 
diff --git a/components/content_settings/core/browser/content_settings_pref_provider.cc b/components/content_settings/core/browser/content_settings_pref_provider.cc
index 74500441b3e9b..a3e2310994721 100644
--- a/components/content_settings/core/browser/content_settings_pref_provider.cc
+++ b/components/content_settings/core/browser/content_settings_pref_provider.cc
@@ -76,8 +76,10 @@ constexpr char kObsoleteTopLevelTpcdOriginTrialExceptionsPref[] =
 // except via enterprise policy, so it is temporarily cleaned up here to revert
 // it to its default value.
 // TODO(https://crbug.com/367181093): clean this up.
+#if !BUILDFLAG(IS_ANDROID)
 constexpr char kBug364820109AlreadyWorkedAroundPref[] =
     "profile.did_work_around_bug_364820109_exceptions";
+#endif  // !BUILDFLAG(IS_ANDROID)
 constexpr char kLocalNetworkAccessMigrateExceptionsPref[] =
     "profile.content_settings.exceptions.has_migrated_local_network_access";
 constexpr char kObsoleteTrackingProtectionExceptionsPref[] =
@@ -129,7 +131,9 @@ void PrefProvider::RegisterProfilePrefs(
       kObsoleteTopLevelTpcdOriginTrialExceptionsPref);
   registry->RegisterDictionaryPref(kObsoleteTrackingProtectionExceptionsPref);
   // TODO(https://crbug.com/367181093): clean this up.
+#if !BUILDFLAG(IS_ANDROID)
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
+#endif  // !BUILDFLAG(IS_ANDROID)
   registry->RegisterBooleanPref(kLocalNetworkAccessMigrateExceptionsPref,
                                 false);
 #endif  // !BUILDFLAG(IS_IOS)
@@ -463,7 +467,9 @@ void PrefProvider::DiscardOrMigrateObsoletePreferences() {
   prefs_->ClearPref(kObsoleteTopLevelTpcdOriginTrialExceptionsPref);
   prefs_->ClearPref(kObsoleteTrackingProtectionExceptionsPref);
   // TODO(https://crbug.com/367181093): clean this up.
+#if !BUILDFLAG(IS_ANDROID)
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
+#endif  // !BUILDFLAG(IS_ANDROID)
 #endif  // !BUILDFLAG(IS_IOS)
 }
 
